src/configs/markdown.ts

```diff
@@ -1,51 +1,77 @@
-import * as parserPlain from 'eslint-parser-plain'
-import { mergeProcessors, processorPassThrough } from 'eslint-merge-processors'
-import type { FlatConfigItem, OptionsComponentExts, OptionsFiles, OptionsOverrides } from '../types'
-import { GLOB_MARKDOWN, GLOB_MARKDOWN_CODE, GLOB_MARKDOWN_IN_MARKDOWN } from '../globs'
+import type { Linter } from 'eslint'
 import { interopDefault } from '../utils'
+import type { FlatConfigItem, OptionsComponentExts, OptionsFiles, OptionsOverrides } from '../types'
+import { GLOB_MARKDOWN_CODE, GLOB_MARKDOWN_OR_MDX } from '../globs'
 
 export async function markdown(
   options: OptionsFiles & OptionsComponentExts & OptionsOverrides = {},
 ): Promise<FlatConfigItem[]> {
   const {
     componentExts = [],
-    files = [GLOB_MARKDOWN],
     overrides = {},
   } = options
 
-  // @ts-expect-error missing types
-  const markdown = await interopDefault(import('eslint-plugin-markdown'))
+  const mdx = await interopDefault(import('eslint-plugin-mdx'))
+  const mdxParser = mdx.configs.flat.languageOptions!.parser!
+
+  const patchedParser: Linter.ParserModule = {
+    ...mdxParser,
+    parse(text, options) {
+      // @ts-expect-error cast
+      const result = mdxParser.parseForESLint!(text, options)
+      const body = result.ast.body
+
+      function predicate(token: { start: number, end: number }) {
+        for (const node of body) {
+          if (node.start <= token.start! && node.end >= token.end!)
+            return true
+        }
+        return false
+      }
+
+      // `eslint-mdx` produces extra tokens that are not presented in the AST, causing rules like `indent` to fail
+      result.ast.tokens = result.ast.tokens.filter(predicate)
+      result.ast.comments = result.ast.comments.filter(predicate)
+
+      return result
+    },
+  }
```

Comment:
According to https://github.com/eslint-stylistic/eslint-stylistic/issues/215

This patch is the closest thing I could fix, but the formatting for mdx is still not ideal.

I think I have spent too much time on working around it. I rather not have it or something in a broken state. I am giving it up at this moment, but I am open to revisiting it later if the compatibility gets better in the future.

Reply:
I don't understand, why not just disable `indent` temporarily for `.md/.mdx` files.

Reply:
https://github.com/antfu/eslint-config/pull/353/commits/b6114b4750ca8fdb6dbd16146d7cc5d91d70fd45

In this commit, even the tests are passing, you see that mdx is basically not formatted at all.

Reply:
Because prettier does not mdx v2 well?

Reply:
<img width="1126" alt="image" src="https://github.com/antfu/eslint-config/assets/8336744/f5dc362e-77c8-40d7-80cc-878c30b07e04">

I tried to run that file locally, it still there are errors reporting but not auto fixed by `eslint-plugin-prettier`. I'm not sure to understand why for now. A deeper debugging is needed.

Reply:
I just raised https://github.com/mdx-js/eslint-mdx/issues/491 for tracking.

Reply:
Ok, looking forward to see it get fixed! 👍